micro_dl/cli/torch_inference_script.py

```diff
@@ -0,0 +1,101 @@
+import os
+import yaml
+import datetime
+import torch
+import micro_dl.inference.image_inference as image_inf
+import micro_dl.torch_unet.utils.inference as torch_inference_utils
+import micro_dl.utils.train_utils as train_utils
+
+import argparse
+
+def read_config(config_path):
+    '''
+    One-line to safely open config files for argument reading
+    
+    :param str config_path: abs or relative path to configuration file
+    
+    :return dict config: a dictionary of config information read from input file
+    '''
+    with open(config_path, 'r') as f:
+        config = yaml.safe_load(f)
+    return config
+
+def parse_args():
+    """
+    Parse command line arguments
+    In python namespaces are implemented as dictionaries
+    
+    :return: namespace containing the arguments passed.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--gpu',
+        type=int,
+        default=None,
+        help=('Optional: specify the gpu to use: 0,1,...',
+              ', -1 for debugging. Default: pick best GPU'),
+    )
+    parser.add_argument(
+        '--gpu_mem_frac',
+        type=float,
+        default=None,
+        help='Optional: specify gpu memory fraction to use',
+    )
+    parser.add_argument(
+        '--config',
+        type=str,
+        help='path to yaml configuration file',
+    )
+    args = parser.parse_args()
+    return args
+
+def check_save_folder(inf_cfig, prep_cfig):
+    '''
+    Helper method to ensure that save folder exists.
+    If no save folder specified in inference_config, force saving in data
+    directory with dynamic name and timestamp.
+    
+    :param pd.dataframe inference_config: inference config file (not) containing save_folder_name
```

Comment:
docstring params don't match those in function definition. Personally I prefer the params in the docstring since they're easier to read and understand.